ports/directxtk/portfile.cmake

```
@@ -8,61 +8,42 @@ vcpkg_from_github(
     HEAD_REF master
 )
 
-IF (TRIPLET_SYSTEM_ARCH MATCHES "x86")
-    SET(BUILD_ARCH "Win32")
-ELSE()
-    SET(BUILD_ARCH ${TRIPLET_SYSTEM_ARCH})
-ENDIF()
-
-if (VCPKG_PLATFORM_TOOLSET STREQUAL "v140")
-    set(VS_VERSION "2015")
-elseif (VCPKG_PLATFORM_TOOLSET STREQUAL "v141")
-    set(VS_VERSION "2017")
-elseif (VCPKG_PLATFORM_TOOLSET STREQUAL "v142")
-    set(VS_VERSION "2019")
+if("xaudio2-9" IN_LIST FEATURES OR VCPKG_TARGET_IS_UWP OR TRIPLET_SYSTEM_ARCH STREQUAL "arm64")
+    set(FEATURE_OPTIONS -DBUILD_XAUDIO_WIN8=OFF -DBUILD_XAUDIO_WIN10=ON)
```

I don't think we should silently ignore the features the user picks if they happen to be on a mismatching; I think you should detect the bad case and fail instead of attempting to fix it.

Why not just translate the features into the defines directly and let the library deal with the potential conflicts itself?

---

You're absolutely right, the best way to handle this is to set xaudio2-9 as dependency for arm and uwp platforms, see strega-nil post here https://github.com/microsoft/vcpkg/pull/13976#pullrequestreview-515132943

Also there's no conflict between those features. If both are on, XAudio 2.9 precede XAudio 2.8.